packages/config-loader/src/lib/transform/include.ts

```diff
@@ -53,7 +54,11 @@ export function createIncludeTransform(
       return { applied: false };
     }
 
-    const includeValue = input[includeKey];
+    const rawIncludedValue = input[includeKey];
+    const substituteResults = await substitute(rawIncludedValue!, baseDir);
+    const includeValue = substituteResults.applied
+      ? substituteResults.value
+      : rawIncludedValue;
```

Comment:
Let's move this down after the string check below, and then check again whether we got `undefined` back from the substitution, and in that case throw a bit more descriptive error.

Reply:
Applied.